airflow/models/taskinstance.py

```diff
@@ -2573,10 +2573,9 @@ def __eq__(self, other):
     def as_dict(self):
         new_dict = dict(self.__dict__)
         for key in ["start_date", "end_date"]:
-            val = new_dict[key]
-            if not val or isinstance(val, str):
-                continue
-            new_dict[key] = val.isoformat()
+            val = new_dict.get(key)
+            if hasattr(val, "isoformat"):
```

Comment:
Would it be better to do a `isinstance(val, datetime.date)`?

Reply:
i would not trust that with all our datetime shenanigans... would you ? :)

Reply:
🤷

Reply:
We'll, maybe all of our date times inherit from that? If so it'd be fine. But if not we might miss stringifying it is the only thing

Reply:
datetime does inherit from date; pendulum DateTime and Date inherits from datetime and date respectively.

Reply:
Ok then, can do